src/platform/datapath_raw.c

```diff
@@ -485,6 +561,75 @@ CxPlatSocketGetParam(
     return QUIC_STATUS_NOT_SUPPORTED;
 }
 
+CXPLAT_THREAD_CALLBACK(CxPlatRouteResolutionWorkerThread, Context)
+{
+    CXPLAT_ROUTE_RESOLUTION_WORKER* Worker = (CXPLAT_ROUTE_RESOLUTION_WORKER*)Context;
+    CXPLAT_DATAPATH* Datapath = Worker->Datapath;
+
+    while (Worker->Enabled) {
+        CxPlatEventWaitForever(Worker->Ready);
+        if (!CxPlatListIsEmptyNoFence(&Worker->Operations)) {
+            CXPLAT_LIST_ENTRY Operations;
+            CxPlatListInitializeHead(&Operations);
+
+            CxPlatDispatchLockAcquire(&Worker->Lock);
+            if (!CxPlatListIsEmpty(&Worker->Operations)) {
+                CxPlatListMoveItems(&Worker->Operations, &Operations);
+            }
+            CxPlatDispatchLockRelease(&Worker->Lock);
+
+            while (CxPlatListIsEmpty(&Operations)) {
+                CXPLAT_ROUTE_RESOLUTION_OPERATION* Operation =
+                    CXPLAT_CONTAINING_RECORD(
+                        CxPlatListRemoveHead(&Operations), CXPLAT_ROUTE_RESOLUTION_OPERATION, WorkerLink);
+                NETIO_STATUS Status =
+                Status = GetIpNetEntry2(&Operation->IpnetRow);
```

Comment:
I think we should have start/end events in this thread so we can track its state in the logs.

Reply:
Will do this in the last iteration to avoid frequent sidecar churn.

Reply:
Added traces